src/store/modules/wallet.js

```diff
@@ -65,6 +65,22 @@ export default {
         for (let addr in state.changeAddresses) {
           Vue.set(state.changeAddresses[addr], 'privKey', cashlib.PrivateKey.fromObject(state.changeAddresses[addr].privKey))
         }
+        for (const utxo of Object.values(state.utxos)) {
+          if (utxo.type === 'p2pkh') {
+            // Does not keep it's privKey with it because we like edge cases.
```

Comment:
Initially the design here was that utxo's house the minimal data required in order to create the private key.

For `p2pkh`-type outputs this is simply the `address` in order to lookup the private key from our wallet (a similar design is taken in Electron Cash), for `stamp` and `stealth`-type outputs this was the constituent ingredients to recreate the private keys (payloadDigests).

This choice was taken to minimize the amount of data stored in the UTXOs on the assumption that the number of UTXOs would be reasonably large.

During our migration to the new stealth private key scheme the decision was made to eagerly calculate and store the private key on receive. The same decision has been made for the stamp keys somewhere down the line.

Giving we have ditched this design, perhaps we should reassess and just store the private key here also.

In light of these of these recent changes we should probably start to employ this technique to prevent massive traversals.

https://medium.com/@jiihu/how-to-improve-performance-of-vuex-store-c9e3cfb01f72